dowhy/causal_refuters/bootstrap_refuter.py

```diff
@@ -47,6 +89,29 @@ def refute_estimate(self, *args, **kwargs):
                                     n_samples=self._sample_size,
                                     random_state=self._random_state )
 
+            if self._chosen_variables is not None:
+                for variable in self._chosen_variables:
+                    
+                    if ('float' or 'int') in new_data[variable].dtype.name: 
+                        new_data[variable] += np.random.randn(self._sample_size) * self._noise
+                    
+                    elif 'bool' in new_data[variable]:
+                        change_mask = self.get_mask_variables()
+                        # Set these values to zero
+                        new_data[variable] *= change_mask
+                        # Sample a binomial distribution and set the value for those datapoints we set to zero in the previous step
+                        new_data[variable] += (1 - change_mask) * np.random.binomial(BootstrapRefuter.DEFAULT_NUMBER_OF_TRIALS,
+                                                                  BootstrapRefuter.DEFAULT_SUCCESS_PROBABILITY,
+                                                                  self._sample_size).astype(bool) 
+                    
+                    elif 'category' in new_data[variable]:
+                        change_mask = self.get_mask_variables()
+                        # Set these values to zero
+                        new_data[variable] *= change_mask

```

Comment:
The change_mask strategy is cool, but less readable. Isn't a simple solution to do this without a change_mask function like this:
1. sample uniform random numbers, probs. 
2. create a vector of changed values from np.random.choice (changed_values).

Then just do,

new_data[variable] = np.where(probs<parameter, changed_values, new_data[variable])

Reply:
There is one problem though, perhaps also in your strategy: what if the same category is selected by random chance, so we end up not changing the category?

Reply:
That is a definite possibility. One way is to take the set difference between the class present and all the classes for each data point. Then we sample the set obtained to get the required data.